modules/couchbase/couchbase.go

```diff
@@ -0,0 +1,593 @@
+		n1ql, _ := c.MappedPort(ctx, QUERY_PORT)
+		n1qlSSL, _ := c.MappedPort(ctx, QUERY_SSL_PORT)
+
+		body["n1ql"] = n1ql.Port()
+		body["n1qlSSL"] = n1qlSSL.Port()
+	}
+
+	if contains(c.config.enabledServices, search) {
+		fts, _ := c.MappedPort(ctx, SEARCH_PORT)
+		ftsSSL, _ := c.MappedPort(ctx, SEARCH_SSL_PORT)
+
+		body["fts"] = fts.Port()
+		body["ftsSSL"] = ftsSSL.Port()
+	}
+
+	if contains(c.config.enabledServices, analytics) {
+		cbas, _ := c.MappedPort(ctx, ANALYTICS_PORT)
+		cbasSSL, _ := c.MappedPort(ctx, ANALYTICS_SSL_PORT)
+
+		body["cbas"] = cbas.Port()
+		body["cbasSSL"] = cbasSSL.Port()
+	}
+
+	if contains(c.config.enabledServices, eventing) {
+		eventingAdminPort, _ := c.MappedPort(ctx, EVENTING_PORT)
+		eventingSSL, _ := c.MappedPort(ctx, EVENTING_SSL_PORT)
+
+		body["eventingAdminPort"] = eventingAdminPort.Port()
+		body["eventingSSL"] = eventingSSL.Port()
+	}
+
+	_, err := c.doHttpRequest(ctx, MGMT_PORT, "/node/controller/setupAlternateAddresses/external", http.MethodPut, body, true)
+
+	return err
+}
+
+func (c *CouchbaseContainer) configureIndexer(ctx context.Context) error {
+	storageMode := "forestdb"
+	if c.config.isEnterprise {
+		storageMode = "memory_optimized"
+	}
+
+	body := map[string]string{
+		"storageMode": storageMode,
+	}
+
+	_, err := c.doHttpRequest(ctx, MGMT_PORT, "/settings/indexes", http.MethodPost, body, true)
+
+	return err
+}
+
+func (c *CouchbaseContainer) waitUntilAllNodesAreHealthy(ctx context.Context) error {
+	var waitStrategy []wait.Strategy
+
+	waitStrategy = append(waitStrategy, wait.ForHTTP("/pools/default").
+		WithPort(MGMT_PORT).
+		WithBasicCredentials(c.config.username, c.config.password).
+		WithStatusCodeMatcher(func(status int) bool {
+			return status == http.StatusOK
+		}).
+		WithResponseMatcher(func(body io.Reader) bool {
+			response, err := io.ReadAll(body)
+			if err != nil {
+				return false
+			}
+			status := gjson.Get(string(response), "nodes.0.status")
+			if status.String() != "healthy" {
+				return false
+			}
+
+			return true
+		}))
+
+	if contains(c.config.enabledServices, query) {
+		waitStrategy = append(waitStrategy, wait.ForHTTP("/admin/ping").
+			WithPort(QUERY_PORT).
+			WithBasicCredentials(c.config.username, c.config.password).
+			WithStatusCodeMatcher(func(status int) bool {
+				return status == http.StatusOK
+			}),
+		)
+	}
+
+	if contains(c.config.enabledServices, analytics) {
+		waitStrategy = append(waitStrategy, wait.ForHTTP("/admin/ping").
+			WithPort(ANALYTICS_PORT).
+			WithBasicCredentials(c.config.username, c.config.password).
+			WithStatusCodeMatcher(func(status int) bool {
+				return status == http.StatusOK
+			}))
+	}
+
+	if contains(c.config.enabledServices, eventing) {
+		waitStrategy = append(waitStrategy, wait.ForHTTP("/api/v1/config").
+			WithPort(EVENTING_PORT).
+			WithBasicCredentials(c.config.username, c.config.password).
+			WithStatusCodeMatcher(func(status int) bool {
+				return status == http.StatusOK
+			}))
+	}
```

Comment:
Same here: this block is not covered by the tests. Could you include them in them?